src/shared/libfido2-util.h

```diff
@@ -14,6 +14,14 @@ typedef enum Fido2EnrollFlags {
         _FIDO2ENROLL_TYPE_INVALID = -EINVAL,
 } Fido2EnrollFlags;
 
+typedef enum COSEAlgorithm {
+        COSE_ALG_RS256 = -257,
+        COSE_ALG_EDDSA = -8,
+        COSE_ALG_ES256 = -7,
+        _COSE_ALG_TYPE_MAX,
+        _COSE_ALG_TYPE_INVALID = -EINVAL,
+} COSEAlgorithm;
+
```

Comment:
Why these are (mostly) redefined? Why not simply use `COSE_ES256` or friends?

If you really want to define an enum, then please use like that
```
typedef enum COSEAlgorithm {
        COSE_ALG_RS256 = COSE_RS256,
        COSE_ALG_EDDSA = COSE_EDDSA,
        COSE_ALG_ES256 = COSE_ES256,
} COSEAlgorithm;
```
We usually adds _MAX and _INVALID, but I think this is not suitable here. As, it is kind of definition, nothing enumerated.

Reply:
The first attempt failed build due to COSE_* macro missing. Possiblely due to lack of libfido2.

Reply:
I guess the following should work without defining the enum.
```
/* libfido2-util.c */
#if HAVE_LIBFIDO2
int parse_fido2_algorithm(const char *s, int *ret) {
        int a;

        assert(s);

        if (streq(s, "ES256"))
                a = COSE_ES256;
        else if(streq(s, "RS256"))
                a = COSE_RS256;
        else if(streq(s, "EDDSA"))
                a = COSE_EDDSA;
        else
                return -EINVAL;

        if (ret)
                *ret = a;
        return 0;
}
#endif
```
```
/* libfido2-util.h*/
#if HAVE_LIBFIDO2
int parse_fido2_algorithm(const char *s, int *ret);
#else
static inline int parse_fido2_algorithm(const char *s, int *ret) {
        return -EOPNOTSUPP;
}
#endif
```

Reply:
But we still need magic value `COSE_ES256` for the deault value for `arg_fido2_cred_alg`, it's that ok?

Reply:
Please also use `#if,else,endif` there.